doc/developer-guides/coding_guidelines.rst

```diff
@@ -2787,3 +2722,418 @@ Compliant example::
+bracket (either a brace or a parenthesis) that follows. This rule applies to the
+keywords `if`, `else`, `for`, `do`, `while`, `switch` and `return`. This rule
+does not apply to the keyword `sizeof`.
+
+Compliant example::
+
+    uint32_t showcase;
+    
+    if (showcase == 0U) {
+        showcase = 32U;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase;
+       
+       if(showcase == 0U){
+           showcase = 32U;
+       }
+
+
+WS-03: Space shall not exist between the function identifier and the following open-parenthesis
+-----------------------------------------------------------------------------------------------
+
+Compliant example::
+
+    size_t entry_size = sizeof(struct vcpuid_entry);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       size_t entry_size = sizeof (struct vcpuid_entry);
+
+
+WS-04: Space shall not exist right after opening brackets and right before closing ones
+---------------------------------------------------------------------------------------
+
+Brackets in this rule refer to parenthesis, braces and squared brackets.
+
+Compliant example::
+
+    size_t entry_size = sizeof(struct vcpuid_entry);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       size_t entry_size = sizeof( struct vcpuid_entry );
+
+
+WS-05: The `*` characters used for pointers shall be right before the function or variable identifiers
+------------------------------------------------------------------------------------------------------
+
+The following cases shall be covered:
+
+a) For declaration of variables of a pointer type, the `*` character shall be
+   right before the variable identifier with no space in between.
+b) For functions whose return value is of a pointer type, the `*` character
+   shall be right before the function identifyer with no spaces in between in the
+   function prototype.
+
+Compliant example::
+
+    uint32_t *showcase_ptr;
+    uint32_t *showcase_func(void);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t* showcase_ptr;
+       uint32_t* showcase_func(void);
+
+
+WS-06: A single space shall exist around binary and ternary operators
+---------------------------------------------------------------------
+
+This rule applies to all binary arithmetic, bit-wise, logical, relational,
+equality and assignment operators, as well as the ternary conditional operator.
+
+Compliant example::
+
+    uint32_t showcase = 32U;
+    
+    showcase = showcase * 2U;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase=32U;
+       
+       showcase=showcase*2U;
+
+
+WS-07: Space shall not exist after unary operator
+-------------------------------------------------
+
+There shall be no space between a unary operator and its operand. This rule
+applies to array subscripting, function calls, member accesses, prefix or
```

Comment:
subscripting and function calls are not unary operators (and are covered by WS-03 and 04)